packages/eslint-plugin-office-addins/test/rules/load-object-before-read.test.ts

```diff
@@ -0,0 +1,80 @@
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import rule from '../../src/rules/load-object-before-read';
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('load-object-before-read', rule, {
+  valid: [ 
+    {
+      code: `
+        var sheetName = 'Sheet1';
+        var rangeAddress = 'A1:B2';
+        var myRange = context.workbook.worksheets.getItem(sheetName).getRange(rangeAddress);  
+        myRange.load('address');
+        context.sync()
+          .then(function () {
+            console.log (myRange.address);   // ok
+          });`
+    },
+    {
+      code: `
+        var property = worksheet.getItem("sheet");
+        property.load('G2');
+        var variableName = property.G2;`
+    },
+    {
+      code: `
+        var selectedRange = context.workbook.getSelectedRange();
+        selectedRange.load('values');
+        if(selectedRange.values === [2]){}`
+    }
+  ],
+  invalid: [
```

Comment:
Should have a test case involving an assignment (and reassignment) instead of just declaration

-  var myRange;
   myRange = context.workbook.worksheets.getSelectedRange();
- var myRange = context.workbook.worksheets.getSelectedRange();
   myRange = context.workbook.worksheets.getItem(sheetName).getRange(rangeAddress);